db/crud.go

```diff
@@ -1166,7 +1187,8 @@ func (db *Database) updateAndReturnDoc(
 			if doc, err = unmarshalDocument(docid, currentValue); err != nil {
 				return
 			}
-			docOut, writeOpts, syncFuncExpiry, err = documentUpdateFunc(doc, currentValue != nil, allowImport)
+			docExists := currentValue != nil
+			docOut, syncFuncExpiry, body, newRevID, storedBody, oldBodyJSON, docSequence, unusedSequences, changedPrincipals, changedRoleUsers, err = db.documentUpdateFunc(docExists, doc, allowImport, docSequence, unusedSequences, callback, expiry)
```

Comment:
As discussed, above, it might be clearer to move docSequence out of the returned variables, and do docSequence=doc.Sequence after line 1191.  That makes it a bit clearer that when calling documentUpdateFunc, docSequence represents a previously allocated sequence, and after calling documentUpdateFunc, docSequence represents the current sequence.